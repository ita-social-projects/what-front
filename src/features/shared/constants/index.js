export * from './proptypes-shapes.js';
