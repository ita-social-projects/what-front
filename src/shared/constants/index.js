export {
  studentGroupShape, studentGroupsStateShape, studentShape,
  studentsStateShape, mentorShape, mentorsStateShape,
  courseShape, coursesStateShape,
} from './proptypes-shapes.js';
