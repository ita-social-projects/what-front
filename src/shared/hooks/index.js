export { useActions } from './use-actions';