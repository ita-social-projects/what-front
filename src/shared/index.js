export * from './hooks';
export * from './api-service';
export * from './routes';
