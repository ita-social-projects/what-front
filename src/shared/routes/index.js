export { Routes } from './routes.js';
export { paths, homepages } from './paths.js';
