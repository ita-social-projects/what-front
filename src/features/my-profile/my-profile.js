import React from 'react';
import { shallowEqual, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { currentUserSelector } from '@/models';
import { paths } from '@/shared';

import { Button } from '@/components';

import styles from './my-profile.scss';

export const MyProfile = () => {
  const { currentUser } = useSelector(currentUserSelector, shallowEqual);

  const history = useHistory();

  return (
    <div className={styles.wrapper}>
      <div className="container card shadow col-10">
        <div className="container pb-2">
          <h3 className="pt-3">My Profile</h3>
          <hr />
          <div className="row mt-3">
            <div className="col-sm-4 font-weight-bold pb-1">First Name:</div>
            <div className="col-sm-8">{currentUser?.first_name}</div>
          </div>
          <hr />
          <div className="row mt-3">
            <div className="col-sm-4 font-weight-bold pb-1">Last Name:</div>
            <div className="col-sm-8">{currentUser?.last_name}</div>
          </div>
          <hr />
          <div className="row mt-3">
            <div className="col-sm-4 font-weight-bold pb-1">Email address:</div>
            <div className="col-sm-8">{currentUser?.email}</div>
          </div>
          <hr />
          <div className="row mb-2">
            <div className="col-sm-6 offset-sm-6">
              <Button className="float-right" onClick={() => (history.push(paths.CHANGE_PASSWORD))}>
                <span>Change password</span>
              </Button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};