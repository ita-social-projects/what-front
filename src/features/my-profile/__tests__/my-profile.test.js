import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter as Router, useHistory } from 'react-router-dom';

import { MyProfile } from '../my-profile';
import { paths } from '@/shared';

jest.mock('react-redux', () => ({
  useSelector: jest.fn().mockReturnValue({
    currentUser: jest.fn().mockReturnValue({
      firstName: 'User',
      lastName: 'User',
      email: 'test.u@gmail.com',
    }),
  }),
}));

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useHistory: jest.fn().mockReturnValue({
    push: jest.fn(),
  }),
}));

describe('My Profile', () => {
  it('should render component', () => {
    const { container } = render(
      <Router>
        <MyProfile />
      </Router>
    );
    const profileContainer = container.getElementsByClassName('container');

    expect(profileContainer).toMatchSnapshot();
  });

  it('should navigate to change password page', async () => {
    const history = useHistory();

    const { getByRole } = render(
      <Router>
        <MyProfile />
      </Router>
    );

    const changePasswordButton = getByRole('button');

    await waitFor(() => {
      fireEvent.click(changePasswordButton);
    });

    expect(history.push).toHaveBeenCalledWith(paths.CHANGE_PASSWORD);
  });
});
