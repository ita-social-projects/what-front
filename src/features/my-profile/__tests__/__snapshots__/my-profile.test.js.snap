// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Profile should render component 1`] = `
HTMLCollection [
  <div
    class="container card shadow col-10"
  >
    <div
      class="container pb-2"
    >
      <h3
        class="pt-3"
      >
        My Profile
      </h3>
      <hr />
      <div
        class="row mt-3"
      >
        <div
          class="col-sm-4 font-weight-bold pb-1"
        >
          First Name:
        </div>
        <div
          class="col-sm-8"
        />
      </div>
      <hr />
      <div
        class="row mt-3"
      >
        <div
          class="col-sm-4 font-weight-bold pb-1"
        >
          Last Name:
        </div>
        <div
          class="col-sm-8"
        />
      </div>
      <hr />
      <div
        class="row mt-3"
      >
        <div
          class="col-sm-4 font-weight-bold pb-1"
        >
          Email address:
        </div>
        <div
          class="col-sm-8"
        />
      </div>
      <hr />
      <div
        class="row mb-2"
      >
        <div
          class="col-sm-6 offset-sm-6 d-flex justify-content-end"
        >
          <button
            class="btn undefined"
            type="button"
          >
            <span>
              Change password
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    class="container pb-2"
  >
    <h3
      class="pt-3"
    >
      My Profile
    </h3>
    <hr />
    <div
      class="row mt-3"
    >
      <div
        class="col-sm-4 font-weight-bold pb-1"
      >
        First Name:
      </div>
      <div
        class="col-sm-8"
      />
    </div>
    <hr />
    <div
      class="row mt-3"
    >
      <div
        class="col-sm-4 font-weight-bold pb-1"
      >
        Last Name:
      </div>
      <div
        class="col-sm-8"
      />
    </div>
    <hr />
    <div
      class="row mt-3"
    >
      <div
        class="col-sm-4 font-weight-bold pb-1"
      >
        Email address:
      </div>
      <div
        class="col-sm-8"
      />
    </div>
    <hr />
    <div
      class="row mb-2"
    >
      <div
        class="col-sm-6 offset-sm-6 d-flex justify-content-end"
      >
        <button
          class="btn undefined"
          type="button"
        >
          <span>
            Change password
          </span>
        </button>
      </div>
    </div>
  </div>,
]
`;
